resources/rest-service/cloudify/install_agent.py

```diff
@@ -0,0 +1,199 @@
+    if os.name == 'nt':
+        return WindowsRunner(logger)
+    else:
+        return LinuxRunner(logger)
+
+
+class Installer(object):
+
+    def __init__(self, logger, runner, cloudify_agent):
+        self.logger = logger
+        self.runner = runner
+        self.cloudify_agent = cloudify_agent
+
+    def install(self):
+        path = tempfile.mkdtemp()
+        try:
+            package_path = os.path.join(path, self.runner.archive_name())
+            self.runner.download(
+                url=self.cloudify_agent['package_url'],
+                destination=package_path)
+            self.runner.extract(package_path, path)
+            agent_config_path = os.path.join(path, 'agent.json')
+            agent_output_path = os.path.join(path, 'output.json')
+            with open(agent_config_path, 'w') as agent_file:
+                agent_file.write(json.dumps(self.cloudify_agent))
+            agent_cmd = self.runner.env_command(path, 'cfy-agent')
+            command = ('{0} install-local'
+                       ' --agent-file {1}'
+                       ' --output-agent-file {2}').format(
+                           agent_cmd,
+                           agent_config_path,
+                           agent_output_path)
+            self.runner.run(command)
+            with open(agent_output_path) as agent_file:
+                return json.load(agent_file)
+        finally:
+            self.runner.rm_dir(path)
+
+
+def _set_package_url(agent):
+    file_server = agent['manager_file_server_url']
+    if agent['windows']:
+        agent_file = 'cloudify-windows-agent.exe'
+    else:
+        distro, _, distro_codename = platform.dist()
+        agent['distro'] = distro.lower()
+        agent['distro_codename'] = distro_codename.lower()
+        agent_file = '{0}-{1}-agent.tar.gz'.format(
+            agent['distro'],
+            agent['distro_codename'])
+    agent['package_url'] = '{0}/packages/agents/{1}'.format(
+        file_server,
+        agent_file)
+
+
+def prepare_cloudify_agent():
+    agent = get_cloudify_agent()
+    agent['windows'] = os.name == 'nt'
+    if 'package_url' not in agent:
+        _set_package_url(agent)
+    return agent
+
+
+def _setup_logger(name):
+    logger_format = '%(asctime)s [%(levelname)s] [%(name)s] %(message)s'
+    logger = logging.getLogger(name)
+    formatter = logging.Formatter(fmt=logger_format,
+                                  datefmt='%H:%M:%S')
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    logger.setLevel(logging.DEBUG)
+    return logger
+
+
+def _return(value, old_agent_version):
+    ctx.returns(value)
+    # Due to bug in celery:
+    if os.name == 'nt' and old_agent_version.startswith('3.2'):
+        from celery import current_task
+        from cloudify.celery import celery
+        celery.backend.mark_as_done(current_task.request.id, value)
```

Comment:
The same issue was mentioned in here:
https://github.com/cloudify-cosmo/cloudify-agent/blob/master/cloudify_agent/app.py#L74
I am not sure what is an exact cause of issue.
I've noticed that for windows all tasks' backends are disabled - event though backend of main celery object is properly configured. So results are not sent back to the rabbitmq.
This is not an issue anymore because new agents are using  workaround with setting backend_url via configuration update rather than constructor.